server/src/storage/data_transfer.rs

```diff
@@ -0,0 +1,399 @@
+    }
+
+    #[test]
+    fn test_non_key_folder_is_not_compressed_file() {
+        let path = ObjectStorePath::from("test/compressed/test.parquet");
+        assert!(DataTransfer::path_is_compressed_file(path).is_none());
+    }
+
+    #[test]
+    fn test_key_folder_without_compressed_folder_is_not_compressed_file() {
+        let path = ObjectStorePath::from("4330327753845164038/test/test.parquet");
+        assert!(DataTransfer::path_is_compressed_file(path).is_none());
+    }
+
+    #[test]
+    fn test_non_parquet_file_is_not_compressed_file() {
+        let path = ObjectStorePath::from("4330327753845164038/compressed/test.txt");
+        assert!(DataTransfer::path_is_compressed_file(path).is_none());
+    }
+
+    #[test]
+    fn test_compressed_file_is_compressed_file() {
+        let path = ObjectStorePath::from("4330327753845164038/compressed/test.parquet");
+        assert_eq!(DataTransfer::path_is_compressed_file(path), Some(4330327753845164038));
+    }
+
+    #[test]
+    fn test_add_compressed_file_into_new_key() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let (_target_dir, mut data_transfer) = create_data_transfer_component(temp_dir.path());
+        let parquet_path = create_compressed_file(temp_dir.path(), "test");
+
+        assert!(data_transfer.add_compressed_file(&KEY, parquet_path.as_path()).is_ok());
+        assert_eq!(data_transfer.compressed_files.get(&KEY).unwrap(), &COMPRESSED_FILE_SIZE);
+    }
+
+    #[test]
+    fn test_add_compressed_file_into_existing_key() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let (_target_dir, mut data_transfer) = create_data_transfer_component(temp_dir.path());
+        let parquet_path = create_compressed_file(temp_dir.path(), "test");
+
+        data_transfer.add_compressed_file(&KEY, parquet_path.as_path()).unwrap();
+        data_transfer.add_compressed_file(&KEY, parquet_path.as_path()).unwrap();
+
+        assert_eq!(data_transfer.compressed_files.get(&KEY).unwrap(), &(COMPRESSED_FILE_SIZE * 2));
+    }
+
+    #[test]
+    fn test_add_non_existent_file() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let path = temp_dir.path().join(format!("{}/compressed", KEY));
+        fs::create_dir_all(path.clone()).unwrap();
+
+        let (_target_dir, mut data_transfer) = create_data_transfer_component(temp_dir.path());
+
+        let parquet_path = path.join("test_parquet.parquet");
+        assert!(data_transfer.add_compressed_file(&KEY, parquet_path.as_path()).is_err())
+    }
+
+    #[test]
+    fn test_transfer_single_file() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let (target_dir, mut data_transfer) = create_data_transfer_component(temp_dir.path());
+        let parquet_path = create_compressed_file(temp_dir.path(), "test");
+
+        data_transfer.add_compressed_file(&KEY, parquet_path.as_path()).unwrap();
+        data_transfer.transfer_data(&KEY).unwrap();
+
```

Comment:
Maybe add an `assert!()` like those used in Line 329 in `test_transfer_multiple_files()` whenever a tests transfers data to ensure the file is removed from the local storage.

Reply:
Fixed.